packages/core/src/utils/query-cache.ts

```diff
@@ -1,7 +1,7 @@
 import {queryCache} from 'react-query'
 
 export interface QueryCacheFunctions<T> {
-  mutate: (newData: T | ((oldData: T) => T)) => void
+  mutate: (newData: T | ((oldData: T | undefined) => T | undefined)) => void
```

Comment:
I think this function should still return `T` instead of undefined, even if `oldData` is undefined. Right?
```suggestion
  mutate: (newData: T | ((oldData: T | undefined) => T)) => void
```

Reply:
We can try. If I did't miss something, this will throw a TypeScript error. Still I do agree, this *should* still return `T`.

Reply:
Okay, I was wrong. Thanks for watching out!